### ISSUE TITLE
Migrations in django-taggit don't work when SOUTH_TESTS_MIGRATE = True (triggered by python_files=*.py)

### ISSUE THREAD

**Issue by gandalfar** [pytest-dev/pytest-django#158]
Fix for issue #128 introduced ability for SOUTH_TESTS_MIGRATE = True, to correctly run south migrations. In this specific case I want to run migrations since it provides data migration for data that I want to use in tests.

I'm trying to integrate it into one my projects that is using `django-taggit` and it crashes with strange south error:

```
(pytest-example)➜  blog git:(master) ./manage.py test
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 2.7.8 -- py-1.4.23 -- pytest-2.6.1
plugins: django
collected 1 items 

web/tests.py E

================================================================================================== ERRORS ==================================================================================================
_______________________________________________________________________________ ERROR at setup of test_ambassador_permission _______________________________________________________________________________

request = <SubRequest '_django_db_marker' for <Function 'test_ambassador_permission'>>

    @pytest.fixture(autouse=True)
    def _django_db_marker(request):
        """Implement the django_db marker, internal to pytest-django
    
        This will dynamically request the ``db`` or ``transactional_db``
        fixtures as required by the django_db marker.
        """
        marker = request.keywords.get('django_db', None)
        if marker:
            validate_django_db(marker)
            if marker.transaction:
                request.getfuncargvalue('transactional_db')
            else:
>               request.getfuncargvalue('db')

../../../.virtualenvs/pytest-example/src/pytest-django/pytest_django/plugin.py:179: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1337: in getfuncargvalue
    return self._get_active_fixturedef(argname).cached_result[0]
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1351: in _get_active_fixturedef
    result = self._getfuncargvalue(fixturedef)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1403: in _getfuncargvalue
    val = fixturedef.execute(request=subrequest)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1820: in execute
    fixturedef = request._get_active_fixturedef(argname)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1351: in _get_active_fixturedef
    result = self._getfuncargvalue(fixturedef)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1403: in _getfuncargvalue
    val = fixturedef.execute(request=subrequest)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1853: in execute
    self.yieldctx)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/_pytest/python.py:1779: in call_fixture_func
    res = fixturefunc(**kwargs)
../../../.virtualenvs/pytest-example/src/pytest-django/pytest_django/fixtures.py:49: in _django_db_setup
    db_cfg = setup_databases(verbosity=0, interactive=False)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/django/test/runner.py:279: in setup_databases
    verbosity, autoclobber=not interactive)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/south/hacks/django_1_0.py:103: in wrapper
    return f(*args, **kwargs)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/django/db/backends/creation.py:339: in create_test_db
    load_initial_data=False)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/django/core/management/__init__.py:159: in call_command
    return klass.execute(*args, **defaults)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/django/core/management/base.py:285: in execute
    output = self.handle(*args, **options)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/django/core/management/base.py:415: in handle
    return self.handle_noargs(**options)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/south/management/commands/syncdb.py:103: in handle_noargs
    management.call_command('migrate', **options)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/django/core/management/__init__.py:159: in call_command
    return klass.execute(*args, **defaults)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/django/core/management/base.py:285: in execute
    output = self.handle(*args, **options)
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/south/management/commands/migrate.py:111: in handle
    ignore_ghosts = ignore_ghosts,
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/south/migration/__init__.py:179: in migrate_app
    Migrations.invalidate_all_modules()
../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/south/migration/base.py:245: in invalidate_all_modules
    migration.invalidate_module()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Migration: web:0001_initial>

    def invalidate_module(self):
        """
            Removes the cached version of this migration's module import, so we
            have to re-import it. Used when south.db.db changes.
            """
>       reload(self.migration())
E       ImportError: reload(): module web.migrations.0001_initial not in sys.modules

../../../.virtualenvs/pytest-example/lib/python2.7/site-packages/south/migration/base.py:389: ImportError
========================================================================================= 1 error in 0.30 seconds ==========================================================================================
```

I'm not sure if it's a problem in `pytest_django` or `django-taggit`, but my hunch is that you'll probably understand better what's going on with south mocking.

to make it easier to test, I've also created a GitHub repo: https://github.com/gandalfar/pytest-example
that is currently using f10e48d4fb7bca3fa2640f6f1c74c4c10655c76d

**Comment by blueyed** [comment]
@gandalfar 
Can you provide feedback on if the workaround actually worked for you, and/or if you had any other insights?
(I've came across my PR/branch again and just rebased it on pytest-django master)

**Comment by blueyed** [comment]
I've found a way to reproduce it in a test.  The problem is triggered by also using django-taggit: https://github.com/blueyed/pytest_django/commit/5ce816a3e84ba00af19f938151c38d6afa5176d0

The bug is with pytest though, where the PEP 302 importer module breaks `reload`.

I've submitted a PR for pytest at: https://bitbucket.org/pytest-dev/pytest/pull-request/259/fix-reload-for-files-matched-by.

I am closing the issue for pytest-django.
